service/tutorial/order/OrderServices.xml

```diff
@@ -0,0 +1,399 @@
+                <set field="partMap" from="[orderId:part.orderId,facilityId:facilityId]"/>
+                <set field="orderPartDetails" from="orderPartDetails+partMap"/>
+            </iterate>
+        </actions>
+    </service>
+
+    <service verb="get" noun="OrderPartsCount">
+        <description>
+            Find the count of order parts for the customer Party Id, “CustJqp”
+        </description>
+        <out-parameters>
+            <parameter name="orderPartsCount"/>
+        </out-parameters>
+        <actions>
+            <entity-find-count entity-name="orderParts" count-field="orderPartsCount">
+                <econdition field-name="customerPartyId" value="CustJqp"/>
+            </entity-find-count>
+        </actions>
+    </service>
+
+    <service verb="get" noun="OrderPartsInfo">
+        <description>
+            Find all unique Order Parts with shipmentMethodEnumId value as “ShtMthGround” and facilityId as “ZIRET_WH”
+        </description>
+        <out-parameters>
+            <parameter name="orderPartsList"/>
+        </out-parameters>
+        <actions>
+            <entity-find entity-name="orderParts" list="orderPartsList" distinct="true">
+                <econdition field-name="shipmentMethodEnumId" value="ShtMthGround"/>
+                <econdition field-name="facilityId" value="ZIRET_WH"/>
+            </entity-find>
+        </actions>
+    </service>
+    <!-- create order here -->
+    <service verb="create" noun="OrderHeader">
+        <description>
+            This service will create the Order_Header detail.
+        </description>
+        <in-parameters>
+            <parameter name="orderName" required="true"/>
+            <parameter name="currencyUomId" default-value="USD"/>
+            <parameter name="salesChannelEnumId"/>
+            <parameter name="statusId" default-value="orderPlaced"/>
+            <parameter name="placedDate" required="true"/>
+            <parameter name="productStoreId"/>
+            <parameter name="approvedDate"/>
+        </in-parameters>
+        <out-parameters>
+            <parameter name="orderId" required="true"/>
+        </out-parameters>
+        <actions>
+            <service-call name="create#mantle.order.OrderHeader" in-map="context" out-map="context"/>
+        </actions>
+    </service>
+    <!-- create orderPart here -->
+    <service verb="create" noun="OrderPart">
+        <description>
+            This service will create OrderPart data
+        </description>
+        <in-parameters>
+            <parameter name="orderId" required="true"/>
+            <parameter name="partName"/>
+            <parameter name="facilityId" required="true"/>
+            <parameter name="shipmentMethodEnumId"/>
+            <parameter name="customerPartyId" required="true"/>
+            <parameter name="item_details" type="List">
+                <parameter name="itemMap" type="Map">
+                    <parameter name="productId"/>
```

Comment:
Can you check if we can add required for the item_details fields?

Reply:
Okay, Ma'am.